### PR TITLE
fix: Google Sheets datasource rehaul

### PR DESCRIPTION
**Pull request by spark-c**
Rehaul of [Google Sheets datasource](https://docs.appsmith.com/reference/datasources/querying-google-sheets) docs page.

Currently Draft PR. Still writing:
- Update heading
- Delete heading

[Notion doc](https://www.notion.so/appsmith/Google-Sheets-5f850691a9dd45138fc0b748f48dfec6)
Github [ticket](https://github.com/appsmithorg/appsmith/issues/16518)

### CODE REVIEW

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 1]
```diff
@@ -1,583 +1,272 @@
...
-Consider enabling the [**Request confirmation before running query**](./../..//core-concepts/data-access-and-binding/querying-a-database/query-settings.md#request-confirmation-before-running-query) setting to help prevent unintentional loss of data.
-:::
-
-### Save and authorize
+Click on the **Save and Authorize** button once you have selected your **Scope**. You'll be directed to a Google Login screen, where you can log into the account whose spreadsheets you’d like to access.
 
-Click on the **Save and Authorize** button once you have selected your desired scope. You'll automatically be navigated to a Google Login screen, where you should select and log-in to the account whose spreadsheets you’d like to access.
-
-On successful login, a screen appears where you are asked to grant Appsmith certain permissions relating to managing your Google Sheets. Be sure to click Allow in order for Google Sheets to allow Appsmith to manage your spreadsheets.
+On a successful login, a screen appears for granting Appsmith permissions for managing your Google Sheets. Click "Allow" to allow Appsmith to manage your spreadsheets.
```
Comment (danciaclara) [inline website/docs/reference/datasources/querying-google-sheets.md:34]:
```suggestion
Click **Allow** to grant the Appsmith app permission to view and edit files based on the scope.
```

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 2]
```diff
@@ -1,583 +1,272 @@
...
-
-## Fetch details
-
-The operation allows you to fetch the details of a given spreadsheet.
-
-<VideoEmbed host="youtube" videoId="elAabPXK_xk" title="Fetch Details of a Spreadsheet" caption="Fetch Details of a Spreadsheet"/>
-
-You can set the parameter values as below:
+<figure>
+  <img src="/img/google-sheets-query-page.png" style={{width: "100%", height: "auto"}} alt="Configuring a query from the query screen."/>
```
Comment (danciaclara) [inline website/docs/reference/datasources/querying-google-sheets.md:39]: Maybe this image should show the dropdown with all the queries?

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 3]
```diff
@@ -1,583 +1,272 @@
...
+  <figcaption align="center"><i>Configuring a query from the query screen.</i></figcaption>
+</figure>
 
-* [Operation](./querying-google-sheets.md#operation) - Fetch details
-* [Entity](./querying-google-sheets.md#entity) - Spreadsheet
-* [Spreadsheet](./querying-google-sheets.md#spreadsheet) - Either select the name of the spreadsheet from the available options, or you can enable the JS label and add the URL of the desired spreadsheet
+You can create [queries](https://docs.appsmith.com/core-concepts/data-access-and-binding/querying-a-database/query-settings) to fetch or write data to your spreadsheets by clicking (**+**) next to **Queries/JS** in the **Explorer** tab and selecting your Google Sheets datasource. You'll be brought to a new screen to set up your query.
```
Comment (danciaclara) [inline website/docs/reference/datasources/querying-google-sheets.md:43]: place the image below this para

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 4]
```diff
@@ -1,583 +1,272 @@
...
-## Insert one
+### Operations
 
-With Insert One, you can choose to insert a new row in an existing spreadsheet or add a new spreadsheet.
+**Operation** sets the type of action you want to perform with your query. A "Spreadsheet" is a document, a "Sheet" is a page of a spreadsheet, and "Sheet Rows" are horizontal records in a sheet.
 
-### Insert a new spreadsheet
+| **Operation**                        | **Description**                           | **Available entities:**       |
+| ------------------------------------ | ----------------------------------------- | ----------------------------- |
+| [**Fetch Details**](#fetch-details)  | Fetches metadata about a spreadsheet. | [Spreadsheet](#fetch-details-spreadsheet)  |
```
Comment (danciaclara) [inline website/docs/reference/datasources/querying-google-sheets.md:51]: Mention this is only accessible if you give Appsmith the read-write All permission

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 5]
```diff
@@ -1,583 +1,272 @@
...
-<VideoEmbed host="youtube" videoId="eSZyxR3vAj4" title="Insert a Spreadsheet" caption="Insert a Spreadsheet"/>
+| **Configuration Field**     | **Description**                                                               |
+| --------------------------- | ----------------------------------------------------------------------------- |
+| **Entity**                  | Select which entity type you want to query. (Sheet Rows, Spreadsheet, Sheet). |
+| **Spreadsheet**             | Select which spreadsheet you want to query from.                              |
+| **Sheet Name**              | Select which sheet you want to query from the spreadsheet.                    |
+| **Table Heading Row Index** | Provide the index of the row in the spreadsheet that contains the headings or labels for your table columns. The first row of the spreadsheet is row 1.                                                     |
 
-For creating a spreadsheet and adding data to it, you can set the parameter values as below:
+### Fetch details
```
Comment (danciaclara) [inline website/docs/reference/datasources/querying-google-sheets.md:68]:
```suggestion
### Fetch Details
```

Let's keep the text as it appears on the product

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 6]
```diff
@@ -1,583 +1,272 @@
...
-    "rowIndex" : 2,
-    "id": "EMP002",
-    "FullName": "Darren L",
-    "Experience": "10.5 years",
-    "Designation": "Senior Manager - Sales"
-    "JoiningDateTime": "2012-05-12T09:07:57.000Z"
-  }
-]
-```
+##### Spreadsheet {#fetch-details-spreadsheet}
```
Comment (danciaclara) [inline website/docs/reference/datasources/querying-google-sheets.md:72]: As this is the only option, a heading here is not required

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 7]
```diff
@@ -1,583 +1,272 @@
...
-```
-{
-	"id": {{Employee.data.EmployeeCode}},
-	"FullName" : {{Employee.data.NameOfEmployee}},
-	"Experience" : {{Employee.data.ExperienceInYears}},
-	"Designation" : {{Employee.data.EmployeeDesignation}},
-	"JoiningDateTime" : {{Employee.data.EmployeeDOJ}}
-}
-```
+To limit the amount of records you receive at once, use **Pagination Limit**. **Pagination Offset** allows skipping a given number of records before returning results; these two fields together enable you to implement pagination for large datasets.
```
Comment (danciaclara) [inline website/docs/reference/datasources/querying-google-sheets.md:99]:
```suggestion
To limit the amount of records you receive at once, use **Pagination Limit**. **Pagination Offset** allows skipping a given number of records before returning results; these two fields enable you to implement [serverside pagination](/reference/widgets/table#server-side-pagination) for large datasets.
```

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 8]
```diff
@@ -1,583 +1,272 @@
...
 
-:::info
-You can choose not to provide **rowIndex**, and inserts still happen.
-:::
-
-You can also bind an API or a Query response returning a JSON Array to Row Objects by using `{{<QUERY/API_NAME>.data}}` embedded in a mustache `{{}}` sign.
+When your query is executed, the new record is inserted as the new highest index in your dataset (at the bottom of the spreadsheet).
 
-## Update many
+#### Insert many
```
Comment (danciaclara) [inline website/docs/reference/datasources/querying-google-sheets.md:193]:
```suggestion
### Insert Many
```

Let it be same heading level

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 9]
```diff
@@ -1,583 +1,272 @@
...
-* [Insert One](./querying-google-sheets.md#insert-one)
-* [Insert Many](./querying-google-sheets.md#insert-many)
-* [Update One ](./querying-google-sheets.md#update-one)
-* [Update Many](./querying-google-sheets.md#update-many)
-
-### Read, edit, create, and delete files
+The **Scope** setting defines what type of privileges your app has while querying your spreadsheets. Use this option to allow the minimum necessary privileges for your app's function to reduce the risk to your data.
 
-To delete spreadsheets, you are required to authorize the Read, Edit, Create, and Delete scope for your datasource.
+- **Read/Write | Selected Google Sheets**: Your app only has read and write access to the spreadsheets that it creates. Spreadsheets on the Google account created by other means aren't visible or mutable by your queries.
```
[the remark this answers is not in the document]
Comment (spark-c) [reply website/docs/reference/datasources/querying-google-sheets.md:19]: > I don't think it has any relation to creation of spreadsheet.

Unfortunately, with the "Selected sheets" scope, I don't think there is any way to access a spreadsheet that your app has not created itself.

File: website/docs/reference/datasources/querying-google-sheets.md [hunk 10]
```diff
@@ -1,583 +1,272 @@
...
 
-The **Delete** permission is required for the following query type:
+- **Read/Write | All Google Sheets**: Your app has read and write access to all spreadsheets on your Google account, regardless of how or when they were created.
 
-* [Delete One](./querying-google-sheets.md#delete-one)
+- **Read Files | All Google Sheets**: Your app has read-only access to all spreadsheets on your Google account, regardless of how or when they were created.
 
-:::caution
-When configured with this scope, queries related to this datasource may overwrite existing records and should be used with care to avoid mistakenly deleting data.
+#### Save and authorize
```
[the remark this answers is not in the document]
Comment (spark-c) [reply website/docs/reference/datasources/querying-google-sheets.md]: I don't quite remember the details of reasoning, but this came up in the weekly review and we kept it; I think it was to make it extra clear that the user **must** use this button and go through the flow or the datasource will not be saved.

File: website/docs/reference/datasources/querying-google-sheets.md (hunk 2 above)
Comment (spark-c) [reply website/docs/reference/datasources/querying-google-sheets.md:34]: that section was removed, so the image was moved somewhere that makes more sense for what is shown

**Review by danciaclara — APPROVED** [review]

**Pull request merged**